### ISSUE TITLE
Reader is affecting javascript on page

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#4266]
**Reported by jstafford on 2014-07-11 21:18**
When NVDA is active, it seems to be stealing focus from the page. For example, when the datePicker (https://github.com/freqdec/datePicker) is opened, I should be able to navigate the calendar with the arrow keys. If I turn off NVDA, then it works fine. The second I turn it back on, I cannot navigate. Their datePicker seems to rely on the sequential focus (i.e. when you open the calendar, the grid is auto-focused so that you can navigate, but if you do something to affect that focus, it becomes un-navigable) so my first thought is that NVDA may be doing something to steal focus causing a break in the control.

I realize this could be directed at the developer of the other product, however, if the defect is caused by NVDA stealing focus or something of the like, then it seems to be a bug with the reader.

**Comment by ehollig** [comment]
Pretty old issue but:
@jcsteh commented on [Jul 12, 2014, 3:43 AM MDT](https://github.com/nvaccess/nvda/issues/4266#issuecomment-155321866 "2014-07-12T09:43:57Z - Replied by Github Reply Comments"):
> Comment 1 by jteh on 2014-07-12 09:43
> Focus in [their demo](http://freqdec.github.io/datePicker/demo/) seems to work as expected for me, although:
> 1. It doesn't seem to respond properly to the button being clicked. I had to specifically pass the enter key through (NVDA+f2 then enter).

This is still true using Firefox 57 and latest NVDA.

> 2. It seems to be very slow.

This is only the case when navigating from one month to the next. It appears the next month has to load to be visible on the screen before NVDA responds.

> 3. I get some double speaking. I'm not quite sure why this is yet.

Not the case using Firefox 57 and latest NVDA Master.

Any ideas what to do with this old issue? I think the issue is now only dealing with point 1 above. CC @feerrenrut @jcsteh @michaelDCurran

**Comment by feerrenrut** [comment]
I tested this with "Demo 1" on [their demo](http://freqdec.github.io/datePicker/demo/) with Firefox 57.
1. While in browse mode, I could tab to the button. NVDA reads "Show Calendar menu button  subMenu  clickable  Show Calendar"
1. Then pressing `Enter` and the calendar is shown, however NVDA does not read anything.
1. Pressing the the keyboard shortcuts, nothing happens.

If focus mode is set manually when focus is on the "Date" edit control then I can use the keyboard shortcuts.
1. Tab to the "Date" edit control
1. Press `NVDA+space` to toggle to browse mode then back to focus mode.
1. Press Tab to get to the "show calendar menu button" and press enter
1. NVDA reads the following:
`Date:  KEYBOARD CONTROLS Tab to the activation button and press Enter to launch a keyboard aware datepicker control. table November 16, 2017  thu  row 7  thu  column 4`
1. Now use arrow keys etc to interact with the datepicker.

**Comment by Adriani90** [comment]
@jstafford, @joyadeponche, @ehollig can you still reproduce this issue?
@jcsteh could this also be a Firefox issue?

**Comment by jstafford** [comment]
This issue is so old I don't remember all of the context around it. I think I was trying to get the single page web application I was developing to work with NVDA, but when NVDA didn't seem to work right for me on other simple web pages, I filed this bug of a specific case I ran into and moved onto other things. I don't have easy access to a Windows machine right now (doing all my development on MacOS), so I can't retest this, but as far as I'm concerned you can close this bug whether it is fixed or not. Sorry I can't be of more help.

**Comment by ehollig** [comment]
Closing as requested by the original author.

**Comment by Adriani90** [comment]
@jstafford feel free to open a new issue if it comes up again. Then we can reopen this one. Thanks for reporting.
